stdlib/public/core/UnsafeRawPointer.swift

```diff
@@ -455,12 +455,27 @@ public struct UnsafeRawPointer: _Pointer {
   /// - Returns: A new instance of type `T`, read from the raw bytes at
   ///   `offset`. The returned instance isn't associated
   ///   with the value in the range of memory referenced by this pointer.
+  @inlinable
   @_alwaysEmitIntoClient
   public func loadUnaligned<T>(
     fromByteOffset offset: Int = 0,
     as type: T.Type
   ) -> T {
     _debugPrecondition(_isPOD(T.self))
+#if arch(x86_64)
+    if MemoryLayout<T>.size >= 16 && _isDebugAssertConfiguration() {
+      return withUnsafeTemporaryAllocation(of: T.self, capacity: 1) {
```

Comment:
This is wild. This actually optimizes out?

Reply:
@atrick checked and LLVM eliminates the temporary allocation! It'll be even better when we just eliminate the workaround.

Reply:
@Catfish-Man for the skeptical:

With this PR, this code:
```
func testSIMD16(_ rawBytes: UnsafeRawPointer) -> SIMD16<UInt8> {
  rawBytes.loadUnaligned(as: SIMD16<UInt8>.self)
}
```
generates this IR after LLVM optimization (temp allocation eliminated):
```
define hidden swiftcc <16 x i8> @"$s4main10testSIMD16ys0C0Vys5UInt8VGSVF"(i8* %0) #1 {
entry:
  %1 = bitcast i8* %0 to %Ts6SIMD16Vys5UInt8VG*
  %._storage = getelementptr inbounds %Ts6SIMD16Vys5UInt8VG, %Ts6SIMD16Vys5UInt8VG* %1, i32 0, i32 0
  %._storage._value = getelementptr inbounds %Ts5UInt8V13SIMD16StorageV, %Ts5UInt8V13SIMD16StorageV* %._storage, i32 0, i32 0
  %2 = load <16 x i8>, <16 x i8>* %._storage._value, align 1
  ret <16 x i8> %2
}
```
emitting
```
_$s10testsimd1610testSIMD16ys0C0Vys5UInt8VGSVF:
0000000100003e80        pushq   %rbp
0000000100003e81        movq    %rsp, %rbp
0000000100003e84        movups  (%rdi), %xmm0
0000000100003e87        popq    %rbp
0000000100003e88        retq
```

Reply:
Love it :D (well, the stack frame is a little tragic, but unrelated)